### ISSUE TITLE
[MODULE] Implement mcmc/hmc_sampler.py — blackjax backend

### ISSUE THREAD

**Issue by sambit-giri** [sambit-giri/PSI#15]
## Summary
Implement a Hamiltonian Monte Carlo sampler in `mcmc/hmc_sampler.py` using `blackjax` as the backend.

HMC is particularly efficient for high-dimensional, differentiable posteriors and complements the emcee and nested sampling backends already planned.

## Interface
```python
sampler = HMCSampler(log_prob_fn, step_size=0.01, n_leapfrog=10, backend="blackjax")
sampler.run(theta_init, n_steps=1000, n_warmup=500)
chain = sampler.get_chain()
```

## Tasks
- [ ] Implement `HMCSampler` wrapping `blackjax.hmc` (and optionally `blackjax.nuts`)
- [ ] Handle JAX/numpy array conversion transparently
- [ ] Add warmup / dual averaging step size adaptation
- [ ] Add `get_chain`, `summary` methods
- [ ] Add unit test with Gaussian log-prob
